sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/container/ThroughputContainerController.java

```diff
@@ -0,0 +1,321 @@
+                }
+            })
+            .flatMap(groupController -> {
+                if (groupController.v != null) {
+                    return groupController.v.processRequest(request, originalRequestMono);
+                }
+
+                return originalRequestMono;
+            });
+    }
+
+    public String getTargetContainerRid() {
+        return this.targetContainerRid;
+    }
+
+    @Override
+    public boolean canHandleRequest(RxDocumentServiceRequest request) {
+        checkNotNull(request, "Request can not be null");
+
+        return StringUtils.equals(this.targetContainerRid, request.requestContext.resolvedCollectionRid);
+    }
+
+    private Mono<ThroughputContainerController> createAndInitializeGroupControllers() {
+        return Flux.fromIterable(this.groups)
+            .flatMap(group -> {
+                ThroughputGroupControllerBase groupController =
+                    this.groupControllers.computeIfAbsent(
+                        group.getGroupName(),
+                        groupName -> ThroughputGroupControllerFactory.createController(
+                                        this.connectionMode,
+                                        this.globalEndpointManager,
+                                        group,
+                                        this.maxContainerThroughput.get(),
+                                        this.partitionKeyRangeCache,
+                                        this.targetContainerRid));
+                return Mono.just(groupController);
+            })
+            .flatMap(groupController -> groupController.init())
+            .then(Mono.just(this));
+    }
+
+    private Flux<Void> refreshContainerMaxThroughputTask(CancellationToken cancellationToken) {
+        checkNotNull(cancellationToken, "Cancellation token can not be null");
+
+        if (this.throughputResolveLevel == ThroughputResolveLevel.NONE) {
+            return Flux.empty();
+        }
+
+        return Mono.delay(DEFAULT_THROUGHPUT_REFRESH_INTERVAL)
+            .flatMap(t -> this.resolveContainerMaxThroughput())
+            .flatMapIterable(controller -> this.groupControllers.values())
+            .doOnNext(groupController -> groupController.onContainerMaxThroughputRefresh(this.maxContainerThroughput.get()))
+            .onErrorResume(throwable -> {
+                //TODO: Figure out how serverless work
+//                if (this.isOfferNotConfiguredException(throwable)) {
+//                    // Throughput is not configured on container nor database, a good hint the resource does not exists any more
+//                    this.close();
+//                }
+
+                logger.warn("Refresh throughput failed with reason %s", throwable);
+                return Mono.empty();
+            })
+            .then()
+            .repeat(() -> !cancellationToken.isCancellationRequested());
+    }
+
+    @Override
+    public Mono<Void> close() {
```

Comment:
This close() will never close the resources, since the usage of this `close()` is not blocking -> https://github.com/Azure/azure-sdk-for-java/pull/18464/files#diff-82ddb520cf7438aeac0aa75f1cbe32bc34429ea4ad9376576b42253452b42cffR175
As no-one is blocking this call -> 

```
if (shouldRefresh) {
                                containerController.close();
                                this.containerControllerCache.refresh(collectionLink, () -> this.createAndInitContainerController(collectionLink));
                                return this.resolveContainerController(collectionLink);
                            }
```

Reply:
yup, forget to subscribed,

will change to containerController.close().subscribeOn(Scehdulers.single).subscribe